docs/src/testing/testing.md

```diff
@@ -41,7 +41,88 @@ test case.
 Starknet Forge will collect tests only from these places:
 
 - any files reachable from the package root (declared as `mod` in `lib.cairo` or its children)
-- files inside the `tests` directory
+- files inside the [`tests`](#the-tests-directory) directory
+
+## The *tests* Directory
+
+### With *tests/lib.cairo*
+
+If there is a `lib.cairo` file in `tests` folder,
+then it is treated as an entrypoint to the `tests` package from which tests are collected.
+
+For example, for a package structured this way:
+
+```shell
+$ tree .
+.
+├── Scarb.toml
+├── tests/
+│   ├── lib.cairo
+│   ├── common/
+│   │   └── utils.cairo
+│   ├── common.cairo
+│   ├── test_contract.cairo
+│   └── not_included.cairo
+└── src/
+    └── lib.cairo
+```
```

Comment:
Can we show a simpler example here? I get what are you trying to showcase, but I wonder if it isn't too confusing.